geonode/layers/utils.py

```diff
@@ -502,14 +502,18 @@ def file_upload(filename, name=None, user=None, title=None, abstract=None,
         if len(regions_resolved) > 0:
             layer.regions.add(*regions_resolved)
 
+    if date is not None:
+        layer.date = date
```

Comment:
It did work, but I took a closer look and converting the provided date into a DateTime object first is definitely the correct way to do this. Good catch! I've fixed it.